Nodejs/Product/TestAdapterNetStandard/NetCoreTestDiscoverer.cs

```diff
@@ -147,10 +119,76 @@ void ProcessFiles(string fileAbsolutePath)
             }
         }
 
-        private void DiscoverTests(HashSet<string> fileList, TestFramework testFramework, ITestCaseDiscoverySink discoverySink, IMessageLogger logger, string nodeExePath, string projectFullPath)
+        private void DiscoverTests(Dictionary<string, HashSet<string>> testItems, FrameworkDiscoverer frameworkDiscoverer, ITestCaseDiscoverySink discoverySink, IMessageLogger logger, string nodeExePath, string projectFullPath)
         {
-            var discoverWorker = new TestDiscovererWorker(projectFullPath, NodejsConstants.ExecutorUri, nodeExePath);
-            discoverWorker.DiscoverTests(fileList, testFramework, logger, discoverySink);
+            foreach (var testFx in testItems.Keys)
+            {
+                var testFramework = frameworkDiscoverer.GetFramework(testFx);
+                if (testFramework == null)
+                {
+                    logger.SendMessage(TestMessageLevel.Warning, $"Ignoring unsupported test framework '{testFx}'.");
+                    continue;
+                }
+
+                var fileList = testItems[testFx];
```

Comment:
Does it matter that these won't be ordered consistently?

Reply:
It doesn't matter.